### PR TITLE
FUSETOOLS2-206 if existing version is found and not up to date, ensur…

### PR DESCRIPTION
**Pull request by bfitzpat**
…e update

Signed-off-by: bfitzpat@redhat.com <bfitzpat@redhat.com>

### CODE REVIEW

**Comment by bfitzpat** [comment]
https://github.com/camel-tooling/vscode-camelk/issues/270
https://issues.redhat.com/browse/FUSETOOLS2-206

**Comment by bfitzpat** [comment]
works in all three cases -- if the kamel cli is missing, it is downloaded; if it is a different version, it is downloaded to match the current version (which wasn't the case before); and if the versions match, it is not downloaded

File: src/installer.ts [hunk 1]
```diff
@@ -145,10 +173,10 @@ export async function installKamel(context: vscode.ExtensionContext): Promise<Er
 
 	let versionToUse = latestversion.result.trim();
 
-	await checkKamelCLIVersion().then((currentVersion) => {
-		if (versionToUse.toLowerCase() === currentVersion.toLowerCase()) {
+	await checkKamelNeedsUpdate(versionToUse).then((needsUpdate) => {
+		if (needsUpdate) {
 			// no need to install, it's already here
```
Comment (apupier) [inline src/installer.ts]: this comment is outdated. In this case, the version needs to be downloaded and updated

**Comment by apupier** [comment]
with this PR i tmeans that now a user cannot provide a specific version of the kamel executable? (to support a specific version of deployment for instance)

**Comment by bfitzpat** [comment]
> with this PR i tmeans that now a user cannot provide a specific version of the kamel executable? (to support a specific version of deployment for instance)

Correct. We will always default to the most recent version supported by the tooling with this change. If you have a suggestion on how to offer an additional state, I'm open to suggestions. The only thing I can think is to offer another preference that essentially says "turn off automatic update of CLI" and works around the version check.

**Comment by bfitzpat** [comment]
> given that it is to fix a bug, it would be nice to add a test at least to cover the case of the detected bug

Test added.

**Comment by bfitzpat** [comment]
Created a new issue to handle adding a way for the user to avoid the auto-upgrade - https://github.com/camel-tooling/vscode-camelk/issues/280

**Pull request merged**